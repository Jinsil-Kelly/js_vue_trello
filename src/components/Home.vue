<template>
  <div>Home {{ msg }}</div>
</template>

<script>
export default {
  props: {
    msg: String
  }
};
</script>

<style lang="scss"></style>
