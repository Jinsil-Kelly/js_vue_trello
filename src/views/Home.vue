<template>
  <div class="home">
    <Home msg="Welcome to Jinsil Home" />
    <router-link to="/b/1">Board1</router-link>
    <router-link to="/b/2">Board2</router-link>
  </div>
</template>

<script>
// @ is an alias to /src
import Home from "@/components/Home.vue";

export default {
  name: "home",
  components: {
    Home
  }
};
</script>
