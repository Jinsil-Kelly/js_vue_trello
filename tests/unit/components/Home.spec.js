import Home from "@/views/Home.vue";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import initialState from "@/store";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("HomeComponent", () => {
  let state;

  const build = () => {
    const wrapper = shallowMount(Home, {
      localVue,
      store: new Vuex.Store({
        state
      })
    });
    return {
      wrapper
    };
  };

  beforeEach(() => {
    jest.resetAllMocks();
    state = { ...initialState };
  });

  it("renders the component", () => {
    // arrange
    const { wrapper } = build();

    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });
});
