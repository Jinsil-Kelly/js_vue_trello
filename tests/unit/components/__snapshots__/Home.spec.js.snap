// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent renders the component 1`] = `
<div class="home">
  <div>
    <div class="home-title">Personal Boards</div>
    <div class="board-list">
      <div class="board-item board-item-new"><a href="" class="new-board-btn">
          Create new board...
        </a></div>
    </div>
    <!---->
  </div>
  <router-link-stub to="/b/1" tag="a" event="click">Board1</router-link-stub>
  <router-link-stub to="/b/2" tag="a" event="click">Board2</router-link-stub>
</div>
`;
